### PR TITLE
Hide automated accounts from Group

### PR DESCRIPTION
**Pull request by mananjadhav**
@deetergp Please review this draft PR. 

### Details
- Added certain consts to create exclude list
- Changed the same for IOU options as well
- 

### Fixed Issues

$ https://github.com/Expensify/App/issues/4486

### Tests
1. Tested creating a new group on all platforms
2. Tested Recent option in the New Group
3. Removed Split Money and Request Money options for excluded emails
4. Removed Request Money on the Compose Button

### QA Steps
1. Go to Home screen and click on the Green FAB
2. Choose New Group
3. It shouldn't show up any excluded emails in the list or the recent options
4. Same has to be checked for Split Money and Request Money
5. In the 1:1 chat between Concierge or Chronos, ensure that the Request Money option is not shown

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web
https://user-images.githubusercontent.com/3069065/130951687-a0dcd407-7f24-4ec3-b60d-196255b2dcfb.mp4

#### Mobile Web
https://user-images.githubusercontent.com/3069065/130951706-bfdebdf5-4247-4bfe-b49d-8964c1f3a84c.mp4

#### Desktop
https://user-images.githubusercontent.com/3069065/130951724-c7d54bb9-c261-4ca8-b4e4-9600b330a9c0.mp4

#### iOS
https://user-images.githubusercontent.com/3069065/130951756-3fd81db3-5bf6-4038-ab29-1888bbe3fa30.mp4

#### Android

https://user-images.githubusercontent.com/3069065/130956835-5fb8cc72-4851-4ac9-a9ea-a4eabfa413ab.mp4

### CODE REVIEW

File: src/pages/NewGroupPage.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ import {getNewGroupOptions, getHeaderMessage} from '../libs/OptionsListUtils';
 import ONYXKEYS from '../ONYXKEYS';
 import styles from '../styles/styles';
 import {fetchOrCreateChatReport} from '../libs/actions/Report';
-import CONST from '../CONST';
+import CONST, {EXLCUDED_GROUP_EMAILS} from '../CONST';
```
Comment (marcaaron) [inline src/pages/NewGroupPage.js:11]: typo

**Review by deetergp — CHANGES_REQUESTED** [review]

**Review by jasperhuangg — CHANGES_REQUESTED** [review]
Few minor changes, but overall looks pretty good, nice job!

File: src/libs/OptionsListUtils.js [hunk 2]
```diff
@@ -569,22 +561,15 @@ function getSearchOptions(
...
 function getNewChatOptions(
     reports,
     personalDetails,
     searchValue = '',
-    {
-        excludeConcierge = false,
-        excludeChronos = false,
-        excludeReceipts = true,
-    } = {},
+    excludeLogins = [],
```
[the remark this answers is not in the document]
Comment (mananjadhav) [reply src/libs/OptionsListUtils.js:571]: Updated

**Comment by mananjadhav** [comment]
PR updated with all the commented  changes.

File: src/CONST.js [hunk 3]
```diff
@@ -399,7 +399,20 @@ const EXPENSIFY_EMAILS = [
     CONST.EMAIL.BILLS, CONST.EMAIL.STUDENT_AMBASSADOR, CONST.EMAIL.QA_TRAVIS,
 ];
 
+const EXCLUDED_GROUP_EMAILS = [
```
Comment (deetergp) [inline src/CONST.js:402]: Can you please put one entry per line in each of these for readability's sake? You'd be my hero if you wanted to do the same to the `EXPENSIFY_EMAILS` array too.

**Review by deetergp — CHANGES_REQUESTED** [review]
Looks good and tests well. I had one minor request regarding making the CONST arrays more readable.

File: src/CONST.js (hunk 3 above)
Comment (mananjadhav) [reply src/CONST.js:402]: I'll do that :) Have you verified if the emails in the list are correctly mapped?

Comment (deetergp) [reply src/CONST.js:402]: I have and they all seem good. Can you add one more address to all three lists? `svfg@expensify.com`

Comment (mananjadhav) [reply src/CONST.js:402]: Noted. Will do.

Comment (mananjadhav) [reply src/CONST.js:411]: Done

**Comment by mananjadhav** [comment]
PR updated

**Review by deetergp — DISMISSED** [review]
Looks good, thanks!

File: src/libs/OptionsListUtils.js [hunk 4]
```diff
@@ -412,17 +410,9 @@ function getOptions(reports, personalDetails, draftComments, activeReportID, {
...
-    if (excludeChronos) {
-        loginOptionsToExclude.push({login: CONST.EMAIL.CHRONOS});
-    }
-
-    if (excludeReceipts) {
-        loginOptionsToExclude.push({login: CONST.EMAIL.RECEIPTS});
-    }
+    _.each(excludeLogins, (excludeLogin) => {
+        loginOptionsToExclude.push({login: excludeLogin});
+    });
```
Comment (jasperhuangg) [inline src/libs/OptionsListUtils.js:415]: `excludeLogin` is kinda weird here, we can just use `login`.

```suggestion
    _.each(excludeLogins, (login) => {
        loginOptionsToExclude.push({login});
    });
```

**Review by jasperhuangg — COMMENTED** [review]
Looks good minus one small change 👍

**Comment by mananjadhav** [comment]
PR Updated

**Review by jasperhuangg — DISMISSED** [review]

**Review by deetergp — DISMISSED** [review]
Looks good 👍

**Comment by jasperhuangg** [comment]
@mananjadhav @deetergp PR had some conflicts so I went ahead and fixed them. Gonna merge this once the tests are done 👍

**Comment by mananjadhav** [comment]
Thanks for resolving conflicts @jasperhuangg. I didn't check it recently. I guess it's safe to merge this now?

**Comment by jasperhuangg** [comment]
@deetergp Actually since I was the last one to make a commit, do you mind re-reviewing + merging if you think everything checks out?

**Review by deetergp — APPROVED** [review]

**Comment by botify** [comment]
This has been deployed to production and is now subject to a 7-day regression period.
If no regressions arise, payment will be issued on 2021-09-17.  :confetti_ball:

**Pull request merged**